__tests__/parse-test.js

```diff
@@ -0,0 +1,204 @@
+
+  test("wrapper", () => {
+    (function testWrapper() {
+      (function testBelowFn() {
+        const err = new Error('something went wrong');
+        const trace = parse(err);
+        expect(trace[0].getFunctionName()).toBe('testBelowFn');
+        expect(trace[1].getFunctionName()).toBe('testWrapper');
+      })();
+    })();
+  });
+
+  test("no stack", () => {
+    const err = { stack: undefined };
+    const trace = parse(err);
+
+    expect(trace).toStrictEqual([]);
+  });
+
+  test("test corrupt stack", () => {
+    const err = {};
+    err.stack =
+  'AssertionError: true == false\n' +
+  '    fuck' +
+  '    at Test.run (/Users/felix/code/node-fast-or-slow/lib/test.js:45:10)\n' +
+  'oh no' +
+  '    at TestCase.run (/Users/felix/code/node-fast-or-slow/lib/test_case.js:61:8)\n';
+
+    const trace = parse(err);
+    expect(trace.length).toBe(2);
+  });
+
+  test("trace braces in path", () => {
+    const err = {};
+    err.stack =
+  'AssertionError: true == false\n' +
+  '    at Test.run (/Users/felix (something)/code/node-fast-or-slow/lib/test.js:45:10)\n' +
+  '    at TestCase.run (/Users/felix (something)/code/node-fast-or-slow/lib/test_case.js:61:8)\n';
+
+    const trace = parse(err);
+    expect(trace.length).toBe(2);
+    expect(trace[0].getFileName()).toBe('/Users/felix (something)/code/node-fast-or-slow/lib/test.js');
+  });
+
+  test("trace without column numbers", () => {
+    const err = {};
+    err.stack =
+  'AssertionError: true == false\n' +
+  '    at Test.fn (/Users/felix/code/node-fast-or-slow/test/fast/example/test-example.js:6)\n' +
+  '    at Test.run (/Users/felix/code/node-fast-or-slow/lib/test.js:45)';
+
+    const trace = parse(err);
+    expect(trace[0].getFileName()).toBe("/Users/felix/code/node-fast-or-slow/test/fast/example/test-example.js");
+    expect(trace[0].getLineNumber()).toBe(6);
+    expect(trace[0].getColumnNumber()).toBeNull();
+  });
+
+  test("compare real with parsed stack trace", () => {
+    var realTrace, err;
+    function TestClass() {
+    }
+    TestClass.prototype.testFunc = function () {
+      realTrace = get();
+      err = new Error('something went wrong');
+    }
+
+    var testObj = new TestClass();
+    testObj.testFunc();
+    var parsedTrace = parse(err);
+
+    realTrace.forEach(function(real, i) {
+      var parsed = parsedTrace[i];
+
+      function compare(method, exceptions) {
+        let realValue = real[method]();
+        const parsedValue = parsed[method]();
+
+        if (exceptions && typeof exceptions[i] != 'undefined') {
+          realValue = exceptions[i];
+        }
+
+        //const realJson = JSON.stringify(realValue);
+        //const parsedJson = JSON.stringify(parsedValue);
+        //console.log(method + ': ' + realJson + ' != ' + parsedJson + ' (#' + i + ')');
+        expect(realValue).toBe(parsedValue);
+      }
+
+      compare('getFileName');
+      compare('getFunctionName', {
+        2: 'Object.asyncJestTest',
+        4: 'new Promise'
```

Comment:
This feels weird, but we are parsing what is passed to us. It probably could be better